apps/studio/components/interfaces/SQLEditor/SQLEditor.utils.ts

```diff
@@ -109,3 +109,43 @@ export const compareAsNewSnippet = (sqlDiff: ContentDiff) => {
     modified: sqlDiff.modified,
   }
 }
+
+// [Joshen] Just FYI as well the checks here on whether to append limit is quite restricted
+// This is to prevent dashboard from accidentally appending limit to the end of a query
+// thats not supposed to have any, since there's too many cases to cover.
+// We can however look into making this logic better in the future
+// i.e It's harder to append the limit param, than just leaving the query as it is
+// Otherwise we'd need a full on parser to do this properly
+export const checkIfAppendLimitRequired = (sql: string, limit: number = 0) => {
+  // Remove lines and whitespaces to use for checking
+  const cleanedSql = sql.trim().replaceAll('\n', ' ').replaceAll(/\s+/g, ' ')
+
+  // Check how many queries
+  const regMatch = cleanedSql.matchAll(/[a-zA-Z]*[0-9]*[;]+/g)
+  const queries = new Array(...regMatch)
+  const indexSemiColon = cleanedSql.lastIndexOf(';')
+  const hasComments = cleanedSql.includes('--')
+  const hasMultipleQueries =
+    queries.length > 1 || (indexSemiColon > 0 && indexSemiColon !== cleanedSql.length - 1)
+
+  // Check if need to auto limit rows
+  const appendAutoLimit =
+    limit > 0 &&
+    !hasComments &&
+    !hasMultipleQueries &&
+    cleanedSql.toLowerCase().startsWith('select') &&
+    !cleanedSql.endsWith('limit') &&
+    !cleanedSql.endsWith('limit;') &&
+    !cleanedSql.match('limit [0-9]*[;]?$')
+  return { cleanedSql, appendAutoLimit }
+}
+
+export const prefixWithLimit = (sql: string, limit: number = 0) => {
```

Comment:
very much a nit but shouldn't this be called `suffixWithLimit`? 😅

Reply:
oh yes hahahaha that is correct